pkg/api/defaults.go

```diff
@@ -117,9 +117,11 @@ func (cs *ContainerService) setOrchestratorDefaults(isUpgrade, isScale bool) {
 		return
 	}
 	o := a.OrchestratorProfile
-	o.OrchestratorVersion = common.GetValidPatchVersion(
-		o.OrchestratorType,
-		o.OrchestratorVersion, isUpdate, a.HasWindows())
+	if o.OrchestratorVersion == "" {
```

Comment:
would this cause issue later if the existing version is not supported?

Shouldn't we error out before calling this?

Reply:
if we want to do that, it's better to be done in RP. That's a behavior change. That means when the version is not supported, even scale is not allowed for VMAS cluster.

Reply:
For creating cluster scenario, RP already block it if the k8s version is not supported.
For updating, this change will also let AKS-E honor the input k8s version when generating template.